sheepdog/transactions/upload/entity_factory.py

```diff
@@ -31,6 +31,10 @@ def create(transaction, doc, config=None):
             #       more helpful debugging
             return UploadEntity(transaction, config)
 
+        # Remove asterisks from dict keys
+        for key in doc:
```

Comment:
`for key in list(doc)` would be safer just in case someone do something else in the loop.